spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/EnvironmentController.java

```diff
@@ -100,32 +102,44 @@ public Environment labelled(@PathVariable String name, @PathVariable String prof
 
 	@RequestMapping("/{name}-{profiles}.properties")
 	public ResponseEntity<String> properties(@PathVariable String name,
-			@PathVariable String profiles) throws IOException {
-		return labelledProperties(name, profiles, null);
+			@PathVariable String profiles,
+			@RequestParam(defaultValue = "false") boolean resolvePlaceholders)
+			throws IOException {
+		return labelledProperties(name, profiles, null, resolvePlaceholders);
 	}
 
 	@RequestMapping("/{label}/{name}-{profiles}.properties")
 	public ResponseEntity<String> labelledProperties(@PathVariable String name,
-			@PathVariable String profiles, @PathVariable String label)
-					throws IOException {
+			@PathVariable String profiles, @PathVariable String label,
+			@RequestParam(defaultValue = "false") boolean resolvePlaceholders)
+			throws IOException {
 		validateNameAndProfiles(name, profiles);
-		Map<String, Object> properties = convertToProperties(
-				labelled(name, profiles, label));
-		return getSuccess(getPropertiesString(properties));
+		Environment environment = labelled(name, profiles, label);
+		Map<String, Object> properties = convertToProperties(environment);
+		String propertiesString = getPropertiesString(properties);
+		if (resolvePlaceholders) {
+			propertiesString = resolvePlaceholders(environment, propertiesString);
+		}
+		return getSuccess(propertiesString);
 	}
 
 	@RequestMapping("{name}-{profiles}.json")
 	public ResponseEntity<Map<String, Object>> jsonProperties(@PathVariable String name,
-			@PathVariable String profiles) throws Exception {
-		return labelledJsonProperties(name, profiles, null);
+			@PathVariable String profiles,
+			@RequestParam(defaultValue = "false") boolean resolvePlaceholders)
+			throws Exception {
+		return labelledJsonProperties(name, profiles, null, resolvePlaceholders);
 	}
 
 	@RequestMapping("/{label}/{name}-{profiles}.json")
 	public ResponseEntity<Map<String, Object>> labelledJsonProperties(
 			@PathVariable String name, @PathVariable String profiles,
-			@PathVariable String label) throws Exception {
+			@PathVariable String label,
+			@RequestParam(defaultValue = "false") boolean resolvePlaceholders)
+			throws Exception {
 		validateNameAndProfiles(name, profiles);
-		Map<String, Object> properties = convertToMap(labelled(name, profiles, label));
+		Environment environment = labelled(name, profiles, label);
+		Map<String, Object> properties = convertToMap(environment);
```

Comment:
I didn't do json because it wasn't one big string. Two options, 1) convert to `String` and resolve like yaml or 2) resolve individual values in `convertToMap->convertToProperties`.